### PR TITLE
Fix mis-cased apiVersion causing php8.x fails in SavedSearchTest

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
I also removed the delete test - it really isn't adding anything

### CODE REVIEW

File: tests/phpunit/api/v3/SavedSearchTest.php [hunk 1]
```diff
@@ -166,28 +163,13 @@ public function testSavedSearchIsDeletedWhenSmartGroupIs($apiVersion): void {
...
-
   /**
    * @return array
-   * @throws \CRM_Core_Exception
    */
   protected function setupContactInSmartGroup(): array {
     $result = $this->callAPISuccess('Contact', 'create', [
       'first_name' => 'Joe',
-      'last_name' => 'Schmoe',
+      'last_name' => 'Schmidt',
```
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/SavedSearchTest.php:172]: So we don't have to add 'Shmoe' to our spelling checkers...

Interesingly I had to add the name from the class copyright to my IDE - but I wasn't sure if the copyright really should have been to an individual (from 2015)

Comment (colemanw) [reply tests/phpunit/api/v3/SavedSearchTest.php:172]: @eileenmcnaughton how about this? #26822

**Comment by seamuslee001** [comment]
@eileenmcnaughton test fail relates not but sure why

**Comment by eileenmcnaughton** [comment]
Because the change caused it to start testing apiv4 - I will see how valid the test is

**Comment by eileenmcnaughton** [comment]
I removed the version dataprovider from the test. It never worked, cos it was wrong and then it didn't work cos more work had to be done since apiv3 params were being passed - I think apiv4 has it's own tests now so it is gone

**Pull request merged**